app/src/main/java/com/practicum/playlistmaker/presentation/search/SearchActivity.kt

```diff
@@ -45,49 +43,31 @@ class SearchActivity : AppCompatActivity() {
         binding.historyRecyclerView.layoutManager = LinearLayoutManager(this)
         binding.historyRecyclerView.adapter = historyAdapter
 
-        // ui listeners
         binding.clearButton.setOnClickListener { binding.searchEditText.text?.clear() }
-        binding.clearHistoryButton.setOnClickListener { vm.onClearHistory() }
+        binding.clearHistoryButton.setOnClickListener { viewModel.onClearHistory() }
 
         binding.searchEditText.setOnEditorActionListener { _, actionId, _ ->
             if (actionId == EditorInfo.IME_ACTION_SEARCH || actionId == EditorInfo.IME_ACTION_DONE) {
                 hideKeyboard()
-                vm.onQueryChanged(binding.searchEditText.text.toString())
+                viewModel.onQueryChanged(binding.searchEditText.text.toString())
                 true
             } else false
         }
 
         binding.searchEditText.addTextChangedListener(SimpleTextWatcher { text ->
             binding.clearButton.isVisible = text.isNotEmpty()
-            vm.onQueryChanged(text)
+            viewModel.onQueryChanged(text)
         })
 
-        binding.retryButton.setOnClickListener { vm.onRetry() }
+        binding.retryButton.setOnClickListener { viewModel.onRetry() }
 
-        // observe state
-        vm.state.observe(this) { st ->
+        viewModel.state.observe(this) { st ->
             when (st) {
-                is SearchState.Loading -> {
-                    setAllGone(); binding.progressBar.visibility = View.VISIBLE
-                }
-                is SearchState.Content -> {
-                    setAllGone()
-                    binding.tracksRecyclerView.visibility = View.VISIBLE
-                    tracksAdapter.setData(st.items)
-                }
-                is SearchState.Empty -> {
-                    setAllGone(); binding.emptyPlaceholder.visibility = View.VISIBLE
-                }
-                is SearchState.Error -> {
-                    setAllGone(); binding.errorPlaceholder.visibility = View.VISIBLE
-                }
-                is SearchState.History -> {
-                    setAllGone()
-                    if (st.items.isNotEmpty()) {
-                        binding.historyGroup.visibility = View.VISIBLE
-                        historyAdapter.setData(st.items)
-                    }
-                }
+                is SearchState.Loading -> { setAllGone(); binding.progressBar.visibility = View.VISIBLE }
```

Comment:
Можно использовать котлиновский экстенш `isVisible` для управления видимостью вьюшек:

```suggestion
                is SearchState.Loading -> { setAllGone(); binding.progressBar.isVisible = true }
```